### PR TITLE
chore: Migrate doctor command to TypeScript (3)

### PR DESCRIPTION
**Pull request by thib92**
Summary:
---------

Finish the healthchecks migration to TS, continuing the work from #712 and #716 (linked to #683):

* `androidSDK`
* `iosDeploy`
* `nodeJs`
* `packageManagers`
* `watchman`

Test Plan:
----------

- [x] CI is green ✅ 
- [ ] The changes made make sense

### CODE REVIEW

**Comment by thib92** [comment]
Tests are failing because of the `androidSdk` healthcheck, because here:

https://github.com/thib92/react-native-cli/blob/2ff26561138c1a7c4e71687dc0a12d4f00f20b9b/packages/cli/src/commands/doctor/healthchecks/androidSDK.ts#L20

We check for `sdks === 'Not Found'`, but it is never the case as the typings describe them as:

```typescript
'Android SDK': {
  'API Levels': string[];
  'Build Tools': string[];
  'System Images': string[];
  'Android NDK': string;
};
```

@lucasbento do you know if `'Not Found'` is actually a possible value for the `Android SDK` key?

**Comment by thymikee** [comment]
@thib92 I think `envinfo` may output a string `"Not Found"` when a certain required environment info is not available (looks like it happens on Windows at least, but make sure to check the envinfo codebase or typings)

**Comment by thib92** [comment]
Thanks @thymikee, I’ll take a look!

**Comment by lucasbento** [comment]
> do you know if 'Not Found' is actually a possible value for the Android SDK key?

@thib92: yeah, it is, you can check it out [here](https://github.com/tabrindle/envinfo/blob/master/src/helpers/sdks.js#L62-L65).

**Comment by thib92** [comment]
Thanks. I’ll update the typings then.

File: packages/cli/src/commands/doctor/types.ts [hunk 1]
```diff
@@ -32,12 +32,14 @@ export type EnvironmentInfo = {
     'iOS SDK': {
       Platforms: string[];
     };
-    'Android SDK': {
-      'API Levels': string[];
-      'Build Tools': string[];
-      'System Images': string[];
-      'Android NDK': string;
-    };
+    'Android SDK':
```
Comment (thib92) [inline packages/cli/src/commands/doctor/types.ts]: Ideally, we should contribute these typings to the `envinfo` repo. There are no typings yet in there. There is [an ongoing PR](https://github.com/tabrindle/envinfo/pull/105), but the typings are quite incomplete.

**Comment by thib92** [comment]
CI is failing but doesn't seem related 🤔

**Comment by thib92** [comment]
@thymikee is there a way to "try again" on Azure Pipeline? I ran the E2E tests locally and they worked. It might just be an "infrastructure bug" while it ran the test.

**Comment by Esemesek** [comment]
@thib92 Tests on Windows seem to fail every now and then. I wouldn't care too much C:

File: packages/cli/src/commands/doctor/healthchecks/androidSDK.ts [hunk 2]
```diff
@@ -58,14 +56,8 @@ export default ({
...
   },
-  runAutomaticFix: async ({
-    loader,
-    environmentInfo,
-  }: {
-    loader: typeof Ora,
-    environmentInfo: EnvironmentInfo,
-  }) => {
-    const version = environmentInfo.SDKs['Android SDK'][0];
+  runAutomaticFix: async ({loader, environmentInfo}) => {
```
Comment (Esemesek) [inline packages/cli/src/commands/doctor/healthchecks/androidSDK.ts]: Can you add function parameter type?

File: packages/cli/src/commands/doctor/healthchecks/iosDeploy.ts [hunk 3]
```diff
@@ -20,13 +16,13 @@ const getInstallationCommand = () => {
...
 
-export default ({
+export default {
   label: 'ios-deploy',
   isRequired: false,
   getDiagnostics: async () => ({
     needsToBeFixed: await checkSoftwareInstalled('ios-deploy'),
   }),
-  runAutomaticFix: async ({loader}: {loader: typeof Ora}) => {
+  runAutomaticFix: async ({loader}) => {
```
Comment (Esemesek) [inline packages/cli/src/commands/doctor/healthchecks/iosDeploy.ts]: Missing function parameters type here as well.

File: packages/cli/src/commands/doctor/healthchecks/nodeJS.ts [hunk 4]
```diff
@@ -1,28 +1,23 @@
...
-// $FlowFixMe - converted to TS
 import {logManualInstallation} from './common';
-import type {EnvironmentInfo, HealthCheckInterface} from '../types';
+import {HealthCheckInterface} from '../types';
 
-export default ({
+export default {
   label: 'Node.js',
-  getDiagnostics: async ({Binaries}: EnvironmentInfo) => ({
+  getDiagnostics: async ({Binaries}) => ({
```
Comment (Esemesek) [inline packages/cli/src/commands/doctor/healthchecks/nodeJS.ts]: Here too C:

File: packages/cli/src/commands/doctor/healthchecks/nodeJS.ts [hunk 5]
```diff
@@ -1,28 +1,23 @@
...
     version: Binaries.Node.version,
     needsToBeFixed: doesSoftwareNeedToBeFixed({
       version: Binaries.Node.version,
       versionRange: versionRanges.NODE_JS,
     }),
   }),
-  runAutomaticFix: async ({loader}: {loader: typeof Ora}) => {
+  runAutomaticFix: async ({loader}) => {
```
Comment (Esemesek) [inline packages/cli/src/commands/doctor/healthchecks/nodeJS.ts]: And here C:

**Comment by thib92** [comment]
@Esemesek I guess you figured it out after the comments, but the function types are not needed here because we already have the `HealthCheckInterface`, so everything is implicit 😄

**Comment by Esemesek** [comment]
@thib92 Yup. Figured it out C:

**Pull request merged**